crates/storage/provider/src/post_state.rs

```diff
@@ -764,9 +610,9 @@ mod tests {
         // 0x01 => 1 => 2
         let storage_b_changeset = BTreeMap::from([(U256::from(1), (U256::from(1), U256::from(2)))]);
 
-        post_state.change_storage(address_a, storage_a_changeset);
-        post_state.change_storage(address_b, storage_b_changeset);
-        post_state.write_to_db(&tx, 0).expect("Could not write post state to DB");
+        post_state.change_storage(1, address_a, storage_a_changeset);
```

Comment:
nit - could maybe pull in the blocknumber inside PostState and change_storage wouldn't require it as an explicit arg, but no big deal, we can do in follow up

Reply:
This would make `PostState` have an internal state of block number, this felt shady to me.